File: backend/app/schemas/ranking.py
from pydantic import BaseModel


class RankingItem(BaseModel):
    ranking: int
    player_id: int
    full_name: str
    department: str | None = None
    age: int | None = None
    total_points: int
    tournament_count: int


class PointsDetailItem(BaseModel):
    id: int
    tournament_name: str
    source_type: str
    result_type: str | None = None
    points_earned: int
    description: str | None = None
    tournament_date: str | None = None
    team_name: str | None = None
    team_total_points: int | None = None
    team_member_count: int | None = None


class PointsSummary(BaseModel):
    individual_event: int = 0
    team_share: int = 0
    travel_bonus: int = 0
    representative_team: int = 0
    organizer_bonus: int = 0
    donation_bonus: int = 0


class PlayerPointsResponse(BaseModel):
    player: RankingItem
    summary: PointsSummary
    details: list[PointsDetailItem]

File: frontend/shared/types/ranking.ts
import type { SourceType } from './enums'

export interface RankingItem {
  ranking: number
  player_id: number
  full_name: string
  department?: string
  age?: number
  total_points: number
  tournament_count: number
}

export interface PlayerPointsDetail {
  player: {
    id: number
    full_name: string
    department?: string
    gender?: string
  }
  summary: {
    individual_event: number
    team_share: number
    travel_bonus: number
    representative_team: number
    organizer_bonus: number
    donation_bonus: number
  }
  details: PointsRecord[]
}

export interface PointsRecord {
  id: number
  tournament_name: string
  source_type: SourceType
  points: number
  created_at: string
}
